internal/pagination/books.go

```diff
@@ -0,0 +1,176 @@
+package pagination
+
+import (
+	"bookshop/internal/models"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/govalues/decimal"
+)
+
+const (
+	BookFormatParam   = "format"
+	BookCategoryParam = "category"
+	BookMinPriceParam = "min_price"
+	BookMaxPriceParam = "max_price"
+)
+
+type BookFilters struct {
+	Availability string
```

Comment:
I'd rather stay with just Availability since it's a property of a BookFilters struct, not just a Filters struct. If we were to prefix it with a Book, why just Availability and not all other properties?